modules/standard/Math.chpl

```diff
@@ -1075,8 +1075,8 @@ module Math {
   }
 
   private inline proc chpl_logBasePow2(val: int(?w), baseLog2) {
-    if (val < 1) {
-      halt("Can't take the log() of a non-positive integer");
+    if boundsChecking && val < 1 {
+      HaltWrappers.boundsCheckHalt("Can't take the log() of a non-positive integer");
```

Comment:
I always see `boundsChecking` as a misnomer, because that's how we typically check for `--no-checks`, which is very general and more than bounds checking. So, the line above is something I am numb to, but this line gave me a pause. Is this really bounds checking? I guess `log`'s input has some "bounds", but I feel like I am squinting here. Unfortunately, there doesn't seem to be a proper HaltWrapper for this case.

While I am OK with merging this as-is, I feel a bit queasy. The ideal solution seems to be having a `--math-checks` or something and tying everything together. But I can't convince myself that that worth the time.

Reply:
I can see the concern, but of the flags/global variables available today, `--no-bounds-checks`/`boundsChecking` is the closest match from the list in https://chapel-lang.org/docs/main/usingchapel/man.html#man-checks